src/Check/DirWritable.php

```diff
@@ -24,30 +20,22 @@
  */
 class DirWritable extends AbstractCheck implements CheckInterface
 {
-    /** @var array|Traversable */
+    /** @var non-empty-list<string> */
     protected $dir;
 
     /**
-     * @param  string|array|Traversable $path Path name or an array of paths
+     * @param  string|non-empty-list<string> $path Path name or an array of paths
      * @throws InvalidArgumentException
      */
-    public function __construct($path)
+    public function __construct(string|array $path)
```

Comment:
```suggestion
    public function __construct(string|iterable $path)
```

Traversable was allowed previously